taskana-adapter-camunda-system-connector/src/main/java/pro/taskana/adapter/systemconnector/camunda/config/CamundaSystemConnectorConfiguration.java

```diff
@@ -16,10 +19,16 @@
 /** Configures the camunda system connector. */
 @Configuration
 @DependsOn(value = {"adapterSpringContextProvider"})
+@EnableConfigurationProperties(OkHttpProperties.class)
 public class CamundaSystemConnectorConfiguration {
+
   @Bean
-  public RestTemplate restTemplate(RestTemplateBuilder builder) {
-    return builder.build();
+  RestTemplate restTemplate(RestTemplateBuilder builder, OkHttpProperties okHttpProperties) {
+    return builder
+        .setConnectTimeout(Duration.ofMillis(okHttpProperties.getConnectionTimeout()))
```

Comment:
Why do you add configuration here?

Reply:
Hello @ryzheboka ,

maybe my comment in the commit message was not clear enough. I try to make it better this time. :)

Unfortunately, the default jdk client used so far does not support the HTTP request method PATCH. That's why I switched to the OkHttpClient.

You then need to tell the RestTemplateBuilder which HTTP client to use.

OkHttpClient comes with very large defaults for connection and read timeout. I don't know how to set these timeouts using the default Spring properties. That's why I introduced my own properties (OkHttpProperties) and set them.

I'm always a fan of allowing the user to set the timeouts themselves.

Does this answer your question?

Reply:
> OkHttpClient comes with very large defaults for connection and read timeout. I don't know how to set these timeouts using the default Spring properties. That's why I introduced my own properties (OkHttpProperties) and set them.

Thanks, that was the detail I needed :)